### PR TITLE
DB-11533 Introduce parameter to change return type of count

### PR DESCRIPTION
**Pull request by arnaud-splice**
## Description

This introduces a new parameter `splice.bind.countReturnType` to toggle the return type of COUNT from big int to int.

## How to test

```
select typeof(count(*)) from sysibm.sysdummy1;
1
----------------------------------------
BIGINT

call syscs_util.syscs_set_global_database_property('splice.bind.countReturnType', 'integer');
select typeof(count(*)) from sysibm.sysdummy1;
1
----------------------------------------
INTEGER

call syscs_util.syscs_set_global_database_property('splice.bind.countReturnType', 'bigint');
select typeof(count(*)) from sysibm.sysdummy1;
1
----------------------------------------
BIGINT

### CODE REVIEW

**Review by jyuanca — APPROVED** [review]

**Review by ipraznik-splice — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/SPSPropertyCountReturnType.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import com.splicemachine.db.iapi.types.DataTypeDescriptor;
+import com.splicemachine.db.impl.sql.catalog.Aggregate;
+import com.splicemachine.db.impl.sql.compile.AggregateNode;
+import com.splicemachine.db.impl.sql.compile.CastNode;
+import com.splicemachine.db.impl.sql.compile.ValueNode;
+
+import java.sql.Types;
+
+/**
+ * An SPS property that governs timestamp field(s) in a query.
```
Comment (dgomezferro) [inline db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/SPSPropertyCountReturnType.java:29]: Update comment

**Review by dgomezferro — APPROVED** [review]

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/SPSPropertyCountReturnType.java (hunk 1 above)
Comment (arnaud-splice) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/SPSPropertyCountReturnType.java]: Done

**Comment by OlegMazurov** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2782/

**Review by OlegMazurov — CHANGES_REQUESTED** [review]
Please fix the newly added GlobalDatabasePropertyIT

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/2791/

**Comment by arnaud-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by arnaud-splice** [comment]
> Please fix the newly added GlobalDatabasePropertyIT

done

**Pull request merged**